jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -644,11 +644,29 @@ public Request.Processor handle(Request request) throws Exception
         if (processor != null)
             return processor;
 
-        // The contextRequest is-a Supplier<Processor> that calls effectively calls getHandler().handle(request).
-        // Call this supplier in the scope of the context.
-        Request.Processor contextScopedProcessor = _context.get(contextRequest, contextRequest);
-        // Wrap the contextScopedProcessor with a wrapper that uses the wrapped request
-        return contextRequest.wrapProcessor(contextScopedProcessor);
+        Context lastContext = __context.get();
```

Comment:
Can we use some method in `Context` to do all this instead of doing it here.
Should be something like:
```java
processor = _context.get(() -> getHandler().handle(contextRequest), contextRequest);
```

but I think that one doesn't quite work because it doesn't handle the exception from handle.

Reply:
We were doing that, but it is creating a capturing lambda in a hot path.   So yeah it is a little bit of duplicated behaviour, but it is all in the same code area.  Maybe we need to link with comments so we know where to update if anything changes...

Reply:
I've factored out the duplicate code into protected methods